plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events-v2.ts

```diff
@@ -0,0 +1,554 @@
+}
+
+export type ExportHistoricalEventsUpgradeV2 = Plugin<{
+    global: {
+        _testFunctions: TestFunctions
+    }
+}>
+
+export interface ExportHistoricalEventsJobPayload {
+    // Current cursor to what's being exported
+    timestampCursor: number
+
+    // The lower and upper bound of the timestamp interval to be processed
+    startTime: number
+    endTime: number
+
+    // The offset *within* a given timestamp interval
+    offset: number
+
+    // how many retries a payload has had (max = 15)
+    retriesPerformedSoFar: number
+
+    // used for ensuring only one "export task" is running if the server restarts
+    exportId: number
+
+    // Time frame to fetch events for.
+    fetchTimeInterval: number
+
+    // Key to report export status to
+    statusKey: string
+}
+
+type OffsetParams = Pick<ExportHistoricalEventsJobPayload, 'timestampCursor' | 'fetchTimeInterval' | 'offset'>
+
+export interface ExportHistoricalEventsUIPayload {
+    // Only set starting export from UI
+    parallelism?: number
+    dateFrom: string
+    dateTo: string
+}
+
+export interface ExportParams {
+    id: number
+    parallelism: number
+    dateFrom: ISOTimestamp
+    dateTo: ISOTimestamp
+}
+
+interface CoordinationPayload {
+    running?: Array<ISOTimestamp>
+    done?: Array<ISOTimestamp>
+    progress?: number
+}
+
+interface CoordinationUpdate {
+    hasChanges: boolean
+    done: Array<ISOTimestamp>
+    running: Array<ISOTimestamp>
+    toStartRunning: Array<[ISOTimestamp, ISOTimestamp]>
+    toResume: Array<ExportChunkStatus>
+    progress: number
+    exportIsDone: boolean
+}
+
+export interface ExportChunkStatus extends ExportHistoricalEventsJobPayload {
+    done: boolean
+    progress: number
+    // When was this status recorded
+    statusTime: number
+}
+
+export function addHistoricalEventsExportCapabilityV2(
+    hub: Hub,
+    pluginConfig: PluginConfig,
+    response: PluginConfigVMInternalResponse<PluginMeta<ExportHistoricalEventsUpgradeV2>>
+): void {
+    const { methods, tasks, meta } = response
+
+    const currentPublicJobs = pluginConfig.plugin?.public_jobs || {}
+
+    if (!(INTERFACE_JOB_NAME in currentPublicJobs)) {
+        hub.promiseManager.trackPromise(hub.db.addOrUpdatePublicJob(pluginConfig.plugin_id, INTERFACE_JOB_NAME, {}))
+    }
+
+    const oldRunEveryMinute = tasks.schedule.runEveryMinute?.exec
+
+    tasks.job[INTERFACE_JOB_NAME] = {
+        name: INTERFACE_JOB_NAME,
+        type: PluginTaskType.Job,
+        exec: async (payload: ExportHistoricalEventsUIPayload) => {
+            // only let one export run at a time
+            const alreadyRunningExport = await getExportParameters()
+            if (!!alreadyRunningExport) {
+                createLog('Export already running, not starting another.')
+                return
+            }
+
+            // Clear old (conflicting) storage
+            await meta.storage.del('EXPORT_COORDINATION')
```

Comment:
I did a rename. One is for storing the params, the other running coordination info. Putting all under one increases confusion because one is accessed everywhere, other only for coordination.